### PR TITLE
ci/clang-tidy: make CLANG_TIDY_TARGETS configurable

### PR DESCRIPTION
**Pull request by mathetake**
part of #28566

### CODE REVIEW

**Review by phlax — APPROVED** [review]
lgtm, thanks @mathetake

**Comment by phlax** [comment]
hmm, i wondered if shellcheck would agree

i think you have 2 options - you can either make use of `$@` as suggested before - or to do what you are doing - which i think is better - then you will need to read the env var into an array

File: ci/do_ci.sh [hunk 1]
```diff
@@ -337,11 +337,9 @@ case $CI_TARGET in
         export FIX_YAML="${ENVOY_TEST_TMPDIR}/lint-fixes/clang-tidy-fixes.yaml"
         export CLANG_TIDY_APPLY_FIXES=1
         mkdir -p "${ENVOY_TEST_TMPDIR}/lint-fixes"
-        CLANG_TIDY_TARGETS=(
-            //contrib/...
-            //source/...
-            //test/...
-            @envoy_api//...)
+        TARGETS="${CLANG_TIDY_TARGETS:-//contrib/... //source/... //test/... @envoy_api//...}"
```
Comment (phlax) [inline ci/do_ci.sh:340]:
```suggestion
      if [[ -n "$CLANG_TIDY_TARGETS" ]]; then
          read -ra CLANG_TIDY_TARGETS <<< "${CLANG_TIDY_TARGETS-}"
      else
          CLANG_TIDY_TARGETS=(
            //contrib/...
            //source/...
            //test/...
            @envoy_api//...)
      fi

        TARGETS="${CLANG_TIDY_TARGETS:-//contrib/... //source/... //test/... @envoy_api//...}"
```

File: ci/do_ci.sh [hunk 2]
```diff
@@ -337,6 +337,16 @@ case $CI_TARGET in
         export FIX_YAML="${ENVOY_TEST_TMPDIR}/lint-fixes/clang-tidy-fixes.yaml"
         export CLANG_TIDY_APPLY_FIXES=1
         mkdir -p "${ENVOY_TEST_TMPDIR}/lint-fixes"
+      if [[ -n "$CLANG_TIDY_TARGETS" ]]; then
+          read -ra CLANG_TIDY_TARGETS <<< "${CLANG_TIDY_TARGETS-}"
```
Comment (phlax) [inline ci/do_ci.sh:341]:
```suggestion
          read -ra CLANG_TIDY_TARGETS <<< "${CLANG_TIDY_TARGETS}"
```

apologies, typo

**Comment by mathetake** [comment]
well now it's only running on contrib 🤷

File: ci/do_ci.sh (hunk 2 above)
Comment (phlax) [reply ci/do_ci.sh:341]: also - next 2 lines need to be removed

**Comment by mathetake** [comment]
working now

**Review by phlax — APPROVED** [review]
thanks!

**Pull request merged**